activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -119,7 +134,7 @@ class Array
     alias :orig_sum :sum
 
     def sum(init = nil, &block) #:nodoc:
-      if init.is_a?(Numeric) || first.is_a?(Numeric)
+      if init || first.is_a?(Numeric)
```

Comment:
Ah right, supports other objects if an init arg is provided. It'd be clearer to split up the conditional, then:
```ruby
if init
  orig_sum(init, &block)
elsif first.is_a?(Numeric)
  orig_sum(0, &block)
else
  super
end
```

Since Ruby supports other objects now, we could go even further:
```ruby
if init
  orig_sum(init, &block)
elsif init = _identity_for_sum
  orig_sum(init, &block)
else
  super
end

private def _identity_for_sum #:nodoc:
  case first
  when Numeric; 0
  when String; ''
  when Array; []
  …
  end
end
```